playbooks/etc_hosts/tasks/main.yml

```diff
@@ -0,0 +1,5 @@
+---
+- name: "Build hosts file"
+  lineinfile: dest=/etc/hosts regexp='.*{{ item.replace('.', '-') }}$' line="{{ hostvars[item].ansible_ssh_host }} {{ item.replace('.', '-') }}.{{ etc_hosts_domain }} {{ item.replace('.', '-') }}" state=present
```

Comment:
Looks ok, I agree it'd be better in the long term to drop the usage of dots, but good for now